### PR TITLE
Support Conv+BatchNorm fusion for 1d/3d

### LINKED ISSUE #28757: [feature request] Docs for fuse_conv_bn_eval
Even if it is internal/experimental, it's quite useful for end users as well.

Curently [`fuse_conv_bn_weights`](https://github.com/pytorch/pytorch/blob/master/torch/nn/utils/fusion.py#L22) assumes 2d convolution and hard-codes 4-dim weights, which is not true for Conv1d/Conv3d. However it's easy to support any dimension:
```python
# conv_w = conv_w * (bn_w * bn_var_rsqrt).reshape([-1, 1, 1, 1])
conv_w = conv_w * (bn_w * bn_var_rsqrt).reshape([-1] + [1] * (len(conv_w.shape) - 1))
```

### PR DESCRIPTION
**Pull request by vadimkantorov**
Support Conv+BatchNorm fusion for 1d/3d by being adaptive to number of dimensions (partially fixes https://github.com/pytorch/pytorch/issues/28757)

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by vadimkantorov** [comment]
@soumith Not sure why Facebook Internal check could fail (all other test runs are successful)
